src/lib/Transpose.hs

```diff
@@ -186,8 +186,9 @@ transposeExpr expr ct = case expr of
       False -> do
         e' <- substNonlin e
         void $ buildCase e' UnitTy \i vs -> do
+          vs' <- mapM emitAtomToName vs
```

Comment:
Given that there are places where we do depend on the variables being plain names like this one, wouldn't it be better to just keep the old behavior? It's a change that seems independent from the simplification you want to add anyway, right?

Reply:
`emitAtomToName` checks if it's already a name, so I think there's no loss in this case. The simplifying version of `buildCase` does need to be able to pass atoms rather than names to the continuation, because that's what you get when you successfully unpack the scrutinee (unless you then emit those atoms to obtain names, which *would* be inefficient). So the change isn't independent.